packages/downloads/src/portal-request-dataset-export.ts

```diff
@@ -0,0 +1,208 @@
+import { getUser } from "@esri/arcgis-rest-portal";
+import { RemoteServerError } from './remote-server-error';
+import { urlBuilder } from './utils';
+import { IPortalDatasetExportRequestParameters } from './portal-dataset-export-request-parameters';
+
+
+/**
+ * ```js
+ * import { portalRequestDatasetExport } from "@esri/hub-downloads";
+ * 
+ * const params = {
+ *   host: 'https://hub.arcgis.com/portal,
+ *   datasetId: 'abcdef0123456789abcdef0123456789_0',
+ *   format: 'CSV',
+ *   username: 'user',
+ *   token: '123'
+ * };
+ * portalRequestDatasetExport(params)
+ *   .then(response => {
+ *     // {
+ *     //   downloadId: 'abcdef0123456789abcdef0123456789_0::CSV',
+ *     // }
+ *   });
+ * ```
+ * Request an export of a dataset to a particular file format.
+ * @param params - parameters defining a dataset export job
+ */
+export function portalRequestDatasetExport (params: IPortalDatasetExportRequestParameters) {
```

Comment:
Since this is working directly against portal, I'd expect this can handle private items, so... should we pass in auth/token?